servicetalk-concurrent-api/src/test/java/io/servicetalk/concurrent/api/DefaultExecutorTest.java

```diff
@@ -66,63 +61,132 @@
 import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.junit.jupiter.api.Assumptions.assumeTrue;
 
-@ExtendWith(TimeoutTracingInfoExtension.class)
 public final class DefaultExecutorTest {
 
     private static final int UNBOUNDED = -1;
     private Executor executor;
 
-    public static Stream<Arguments> executors() {
-        // Use new executor for each test per parameter.
-        return Stream.of(
-                newArguments(() -> newFixedSizeExecutor(2), "fixed-size-2", true, 2),
-                newArguments(() -> io.servicetalk.concurrent.api.Executors.newCachedThreadExecutor(), "cached",
-                        true, UNBOUNDED),
-                newArguments(() -> {
-                    ExecutorService service = Executors.newCachedThreadPool();
-                    //noinspection Convert2MethodRef,FunctionalExpressionCanBeFolded
-                    return from(command -> service.execute(command));
-                }, "simple-executor", false, UNBOUNDED),
-                newArguments(() -> from(newScheduledThreadPool(2)), "scheduled",
-                        true,
-                        UNBOUNDED /*Size defines core size, else is unbounded*/),
-                newArguments(() -> from(new ThreadPoolExecutor(2, 2, 60, SECONDS,
-                                new SynchronousQueue<>()), newScheduledThreadPool(2)),
-                        "different-executors", true, 2));
+    private enum ExecutorParam {
+        FIXED_SIZE {
+            @Override
+            boolean supportsCancellation() {
+                return true;
+            }
+
+            @Override
+            Executor get() {
+                return newFixedSizeExecutor(2);
+            }
+
+            @Override
+            int size() {
+                return 2;
+            }
+        },
+        CACHED {
+            @Override
+            boolean supportsCancellation() {
+                return true;
+            }
+
+            @Override
+            Executor get() {
+                return io.servicetalk.concurrent.api.Executors.newCachedThreadExecutor();
+            }
+
+            @Override
+            int size() {
+                return UNBOUNDED;
```

Comment:
No need to change in this PR, but as a hint for the future: sometimes it's a bit more compact to use variables instead of abstract methods when `enum` needs a few options:
https://github.com/apple/servicetalk/blob/627c354947b0ad218eac276ade6b50243414878d/servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/HttpProtocol.java#L27-L38